controller/adr.rst

```diff
@@ -0,0 +1,170 @@
+
+        public function __construct(Environment $twig)
+        {
+            $this->twig = $twig;
+        }
+
+        public function __invoke(): Response
+        {
+            return new Response($this->twig->render('default/index.html.twig'));
+        }
+    }
+
+.. tip::
+
+    As described in the DependencyInjection doc, you must use the __construct() injection
+    approach, this way, your class is easier to update and keep in sync with any framework internal
+    services.
+
+By default, we define the class with the final keyword because this class shouldn't be extended,
+the logic is pretty simple to understand as you understand the ADR pattern, in fact, the 'Action'
+is linked to a single request and his dependencies are linked to this precise Action.
+
+.. tip::
+
+    By using the final approach and the private visibility (inside the container), our class
+    is faster to return and easier to keep out of the framework logic.
+
+Once this is done, you can define the routes like before using multiples approaches:
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        # src/AppBundle/Action/HelloAction.php
+        // ...
+
+        /**
+         * @Route("/hello", name="hello")
+         */
+        final class HelloAction
+        {
+            // ...
+        }
+
+    .. code-block:: yaml
+
+        # app/config/routing.yml
+        hello:
+            path:     /hello
+            defaults: { _controller: AppBundle\Action\HelloAction }
+
+    .. code-block:: xml
+
+        <!-- app/config/routing.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <routes xmlns="http://symfony.com/schema/routing"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/routing
+                http://symfony.com/schema/routing/routing-1.0.xsd">
+
+            <route id="hello" path="/hello">
+                <default key="_controller">AppBundle\Action\HelloAction</default>
+            </route>
+
+        </routes>
+
+    .. code-block:: php
+
+        // app/config/routing.php
+        use AppBundle\Action\HelloAction
+
+        $collection->add('hello', new Route('/hello', array(
+            '_controller' => HelloAction::class,
+        )));
+
+Accessing the request
+---------------------
+
+As you can imagine, as the logic evolve, your class is capable of accessing
+the request from simple method injection like this ::
+
+    <?php
+
+        use Symfony\Component\HttpFoundation\Request;
+        // ...
+
+        public function __invoke(Request $request): Response
+        {
+            return new Response($this->twig->render('default/index.html.twig'));
+        }
+    }
+
+Like you can easily imagine, the :class:`Symfony\\Component\\Httpfoundation\\RequestStack` is the best option to gain access to the request, using this approach, a simple update is recommended and the access to request parameters is way easier::
```

Comment:
Using the `Request` from simple method injection still works in ADR patern, no sure it is the best sample :/